File: backend/src/main/kotlin/com/getamotel/domain/property/RoomType.kt
package com.getamotel.domain.property

import com.getamotel.domain.TenantEntity
import jakarta.persistence.*

@Entity
@Table(name = "room_types")
class RoomType : TenantEntity() {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "property_id", nullable = false)
    lateinit var property: Property

    @Column(nullable = false)
    lateinit var name: String

    @Column(columnDefinition = "TEXT")
    var description: String? = null

    @Column(name = "bed_type")
    var bedType: String? = null

    @Column(name = "max_occupancy", nullable = false)
    var maxOccupancy: Int = 1

    @Column(name = "base_price_cents", nullable = false)
    var basePriceCents: Int = 0

    @Column(name = "total_rooms", nullable = false)
    var totalRooms: Int = 0

    @Column(name = "status", nullable = false)
    @Enumerated(EnumType.STRING)
    var status: RoomTypeStatus = RoomTypeStatus.ACTIVE
}

enum class RoomTypeStatus {
    ACTIVE, INACTIVE
}

File: frontend/lib/features/booking/models/room_type.dart
class RoomType {
  final String id;
  final String name;
  final String? description;
  final String? bedType;
  final int maxOccupancy;
  final int basePriceCents;
  final int totalRooms;

  RoomType({
    required this.id,
    required this.name,
    this.description,
    this.bedType,
    required this.maxOccupancy,
    required this.basePriceCents,
    required this.totalRooms,
  });

  factory RoomType.fromJson(Map<String, dynamic> json) => RoomType(
    id: json['id'],
    name: json['name'],
    description: json['description'],
    bedType: json['bedType'],
    maxOccupancy: json['maxOccupancy'],
    basePriceCents: json['basePriceCents'],
    totalRooms: json['totalRooms'],
  );
}
